source/auth/auth.rst

```diff
@@ -568,6 +568,18 @@ mechanism
 mechanism_properties
 	MUST NOT be specified.
 
+Caching credentials
+`````````````````````
+
+Drivers MUST maintain a cache of computed SCRAM credentials, where each cache
```

Comment:
`MUST` seems strong here, given that [DRIVERS-343](https://jira.mongodb.org/browse/DRIVERS-343) was a "should". Perhaps @behackett feels differently now.

Reply:
Given some feedback I've seen the the support team about the overhead that comes along with SCRAM auth, and since the overhead is about to increase with SCRAM-SHA-256 (slower digest, higher number of iterations), MUST seems appropriate.

Reply:
How many drivers depend on an external SCRAM library (not written by one of us)?  I'd prefer not to force us to implement a library in-house if an external library is sufficient but doesn't cache.

Reply:
From talking to @xdg offline, it sounds like the only two drivers which use external libraries (Perl and Go) both have caching implemented. I'm going to leave this as MUST unless there are any strong objections.